File: original artifact/Ian_Foss_cs-360_project_three_inventory_app/cs360_project_three/app/src/main/java/com/example/cs_360_project_three/InventoryItemDao.java
package com.example.cs_360_project_three;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

// Data access object for inventory items
@Dao
public interface InventoryItemDao {

    @Insert
    long insertItem(InventoryItemEntity inventoryItem);


    @Query("SELECT * FROM inventory_table WHERE name = :name LIMIT 1")
    InventoryItemEntity getItemByName(String name);


    // Returns live data object so grid updates in real time
    @Query("SELECT * FROM inventory_table")
    LiveData<List<InventoryItemEntity>> getAllItems();

    @Update
    void updateItem(InventoryItemEntity item);

    @Delete
    void deleteItem(InventoryItemEntity item);

}

File: enhancements/Enhancement One/CS-499_WIP/app/src/main/java/com/example/CS499_Capstone/InventoryItemDao.kt
package com.example.CS499_Capstone

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

// Data access object for inventory items
@Dao
interface InventoryItemDao {
    @Insert
    fun insertItem(inventoryItem: InventoryItemEntity): Long


    @Query("SELECT * FROM inventory_table WHERE name = :name LIMIT 1")
    fun getItemByName(name: String): InventoryItemEntity


    @get:Query("SELECT * FROM inventory_table")
    val allItems: LiveData<MutableList<InventoryItemEntity>>

    @Update
    fun updateItem(item: InventoryItemEntity)

    @Delete
    fun deleteItem(item: InventoryItemEntity)
}
